google/cloud/storage/transfer_manager.py

```diff
@@ -65,14 +111,9 @@ def upload_many(
 
     :type threads: int
     :param threads:
-        The number of threads to use in the worker pool. This is passed to
-        `concurrent.futures.ThreadPoolExecutor` as the `max_worker`; refer
-        to standard library documentation for details.
-
-        The performance impact of this value depends on the use case, but
-        generally, smaller files benefit from more threads and larger files
-        don't benefit from more threads. Too many threads can slow operations,
-        especially with large files, due to contention over the Python GIL.
+        ***DEPRECATED*** Sets `worker_type` to THREAD and `max_workers` to the
```

Comment:
IIUC, if someone sets a value for this, worker_type automatically changes to "THREAD"? The pydocs specifically mention threads as an efficient solution for smaller files, and the only way to write to file handlers -- but this mentions the param as deprecated. I think this could create some confusion.

Another suggestion is to fully support both (without the deprecation warning) but set WORKERS as the default, or plan to turn down threads altogether.

Reply:
Well, the idea is to grandfather in people who are already using the `threads` keyword argument and ease them into the new system of specifying worker_type and max_workers. So I would encourage people using `threads` to instead use `worker_type` and `max_workers` with the same values. I will clarify the docstring here.